include/envoy/server/instance.h

```diff
@@ -179,6 +179,22 @@ class Instance {
    * @return information about the local environment the server is running in.
    */
   virtual const LocalInfo::LocalInfo& localInfo() PURE;
+
+  /**
+   * @return the flush interval of stats sinks.
+   */
+  virtual std::chrono::milliseconds statsFlushInterval() PURE;
+
+  // TODO (@trabetti) : for the two methods below - too specific? what is a better way?
```

Comment:
Yes, I agree that they're too specific for this.  I have an idea as to how we might be able to get rid of these methods and remove the sink `dynamic_cast`.  

Since the sink receives the `Server::Instance` on construction, you can use the `admin()` method on `Server::Instance` to get a reference to `Admin`.  `Admin` has a `addHandler()` method so you can add the hystrix handler when the sink is constructed (instead of inside the `AdminImpl` constructor).  Then, you can store a reference to the sink inside the handler method (callback), itself.  Let me know if this doesn't make sense, but I think this should also get rid of the requirement that anything outside of the sink know about hystrix.